### PR TITLE
[HOPSWORKS-1537] Drop support for python 2.7 base environments

### PR DESCRIPTION
**Pull request by robzor92**

### CODE REVIEW

File: files/default/sql/ddl/updates/1.3.0.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE `hopsworks`.`conda_environment` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `tf_version` varchar(20) COLLATE latin1_general_cs NOT NULL,
+  `python_version` varchar(20) COLLATE latin1_general_cs NOT NULL,
+  UNIQUE KEY `tensorflow_python_version` (`tf_version`, `python_version`),
+  PRIMARY KEY (`id`)
+) ENGINE=ndbcluster DEFAULT CHARSET=latin1 COLLATE=latin1_general_cs;
+
+ALTER TABLE `hopsworks`.`project` ADD  `conda_environment_id` int(11);
+ALTER TABLE `hopsworks`.`project` ADD FOREIGN KEY `anaconda_environment_fk` (`conda_environment_id`) REFERENCES `hopsworks`.`conda_environment` (`id`) ON DELETE SET NULL ON UPDATE NO ACTION;
```
Comment (SirOibaf) [inline files/default/sql/ddl/updates/1.3.0.sql]: As I said above, I'm not convinced by the new table. But in any case, you should migrate existing environments to the new table structure.

File: Berksfile [hunk 2]
```diff
@@ -3,16 +3,16 @@ source 'https://supermarket.chef.io'
 
 metadata
 
-cookbook 'conda', github: "logicalclocks/conda-chef", branch: "master"
-cookbook 'kagent', github: "logicalclocks/kagent-chef", branch: "master"
+cookbook 'conda', github: "robzor92/conda-chef", branch: "HOPSWORKS-1537"
```
Comment (SirOibaf) [inline Berksfile]: Make sure these are changed before merging

File: files/default/sql/ddl/1.3.0__initial_tables.sql [hunk 3]
```diff
@@ -1016,21 +1016,21 @@ CREATE TABLE `project` (
...
   `archived` tinyint(1) DEFAULT '0',
-  `conda` tinyint(1) DEFAULT '0',
   `logs` tinyint(1) DEFAULT '0',
   `deleted` tinyint(1) DEFAULT '0',
-  `python_version` varchar(25) COLLATE latin1_general_cs DEFAULT NULL,
   `description` varchar(2000) COLLATE latin1_general_cs DEFAULT NULL,
   `payment_type` varchar(255) COLLATE latin1_general_cs NOT NULL DEFAULT 'PREPAID',
   `last_quota_update` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
   `kafka_max_num_topics` int(11) NOT NULL DEFAULT '100',
   `conda_env` tinyint(1) DEFAULT '0',
```
[the remark this answers is not in the document]
Comment (robzor92) [reply files/default/sql/ddl/1.3.0__initial_tables.sql]: conda_env is needed for us to know whether a clone (install or uninstall) or creating an environment from a yml has happened or not. If only conda_environment_id is set but not conda_env then we know that they are still using the base env.

File: files/default/sql/ddl/1.3.0__initial_tables.sql [hunk 4]
```diff
@@ -1488,6 +1488,21 @@ CREATE TABLE `tf_lib_mapping` (
 ) ENGINE=ndbcluster DEFAULT CHARSET=latin1 COLLATE=latin1_general_cs;
 /*!40101 SET character_set_client = @saved_cs_client */;
 
+--
+-- Table structure for table `conda_environment`
+--
+
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `conda_environment` (
```
[the remark this answers is not in the document]
Comment (robzor92) [reply files/default/sql/ddl/1.3.0__initial_tables.sql]: Yeah you are right this introduces a bit of redundancy. Not a huge change to backtrack to the old structure so I might do that.

**Pull request closed without merging**